### PR TITLE
Add version and checksum to generated backup codes

### PR DESCRIPTION
**Pull request by rcchan**

### CODE REVIEW

**Comment by sirdarckcat** [comment]
#378 fixes the CI build

File: src/javascript/crypto/e2e/extension/ui/dialogs/backupkey/backupkey_test.js [hunk 1]
```diff
@@ -73,11 +73,12 @@ function tearDown() {
 
 
 function testOutputVersion() {
-  /* Version bit must be zero, marks assumption that ECDSA + ECDH keys are
+  /* Version must be 1.
+   * Currently marks assumption that ECDSA + ECDH keys are
    * are always generated in pairs by the application */
```
Comment (sirdarckcat) [inline src/javascript/crypto/e2e/extension/ui/dialogs/backupkey/backupkey_test.js]: "are are"

File: src/javascript/crypto/e2e/extension/ui/dialogs/backupkey/backupkey_test.js [hunk 2]
```diff
@@ -87,7 +88,7 @@ function testGetKeyCode() {
   asyncTestCase.waitForAsync('Waiting for dialog to be populated with key');
   dialog.getBackupCode_().addCallback(function(code) {
     assertEquals(code, goog.crypt.base64.encodeByteArray(
-        [1, 1, 2, 3, 4, 5]));
+        [1, 1, 1, 2, 3, 4, 5, 0x8D, 0x87]));
```
Comment (sirdarckcat) [inline src/javascript/crypto/e2e/extension/ui/dialogs/backupkey/backupkey_test.js]: can you add a test shared with e2email?

**Comment by sirdarckcat** [comment]
nice =)

File: src/javascript/crypto/e2e/extension/ui/dialogs/backupkey/backupkey_test.js (hunk 2 above)
Comment (sirdarckcat) [inline src/javascript/crypto/e2e/extension/ui/dialogs/backupkey/backupkey_test.js]: by shared I mean a shared code (so we know it works on both sides)

Comment (rcchan) [inline src/javascript/crypto/e2e/extension/ui/dialogs/backupkey/backupkey_test.js]: It isn't compatible yet. There's no code (on either side) to parse the other.  This simply adds the basic e2e code to e2e.  I can add parsing for e2email codes in a separate PR.

**Pull request merged**